fdbserver/StorageMetrics.actor.cpp

```diff
@@ -349,10 +358,55 @@ void StorageServerMetrics::getStorageMetrics(GetStorageMetricsRequest req,
 	req.reply.send(rep);
 }
 
+std::vector<ReadHotRangeWithMetrics> StorageServerMetrics::getReadHotRanges(KeyRangeRef parentRange,
+                                                                            int splitCount,
+                                                                            uint8_t splitType) const {
+	const StorageMetricSample* sampler;
+	switch (splitType) {
+	case ReadHotSubRangeRequest::SplitType::BYTES:
+		sampler = &byteSample;
+		break;
+	case ReadHotSubRangeRequest::SplitType::READ_BYTES:
+		sampler = &bytesReadSample;
+		break;
+	case ReadHotSubRangeRequest::SplitType::READ_OPS:
+		sampler = &opsReadSample;
+		break;
+	default:
+		ASSERT(false);
+	}
+
+	std::vector<ReadHotRangeWithMetrics> toReturn;
+	double total = sampler->getEstimate(parentRange);
+	double splitChunk = total / splitCount;
```

Comment:
do we want this to be total / (splitCount + 1)? with the current name, split count suggests the number of times to range is split which would produce 1 more range than split points

Reply:
I am sorry I use "split" here leading the concept mix with shard splitting. But I think splitCount means how many chunks we want to have... I should rename it to `chunkCount`